tests/VolleyManagement.Specs/TeamsContext/CreateTeamsSteps.cs

```diff
@@ -1,51 +1,153 @@
         }
-        
-        [Given(@"coach is ")]
-        public void GivenCoachIs()
+
+        [Given(@"captain is (.*)")]
+        public void GivenCaptainIs(string fullName)
         {
-            ScenarioContext.Current.Pending();
+            RegisterNewPlayerAndSetCaptainId(fullName);
+            _captain = new PlayerId(_captainId);
         }
-        
-        [Given(@"achievements are ")]
-        public void GivenAchievementsAre()
+
+        [Given(@"captain empty")]
+        public void GivenCaptainIsEmpty()
         {
-            ScenarioContext.Current.Pending();
+            _captain = null;
         }
-        
+
+        [Given(@"coach is (.*)")]
+        public void GivenCoachIs(string coach)
+        {
+            _coach = coach;
+        }
+
+        [Given(@"achievements are (.*)")]
+        public void GivenAchievementsAre(string achievements)
+        {
+            _achievements = achievements;
+        }
+
         [When(@"I execute CreateTeam")]
         public void WhenIExecuteCreateTeam()
         {
-            ScenarioContext.Current.Pending();
+            try
+            {
+                var teamDto = new CreateTeamDto {
+                    Name = _teamName,
+                    Coach = _coach,
+                    Achievements = _achievements,
+                    Captain = _captain,
+                    Roster = _playerList
+                };
+
+                _team = _teamService.Create(teamDto);
+                _newTeamId = _team.Id;
+            }
+            catch (Exception ex)
+            {
+                _exception = ex;
+            }
         }
-        
+
         [Then(@"new team gets new Id")]
         public void ThenNewTeamGetsNewId()
         {
-            ScenarioContext.Current.Pending();
+            _newTeamId.Should().NotBe(default(int));
+            _newTeamId.Should().NotBe(int.MaxValue);
         }
-        
+
         [Then(@"new team should be succesfully added")]
         public void ThenNewTeamShouldBeSuccesfullyAdded()
         {
-            ScenarioContext.Current.Pending();
+            var teamComparer = new TeamComparer();
+            TeamEntity teamEntity;
+            IEnumerable<PlayerId> roster = new List<PlayerId>();
+            using (var context = TestDbAdapter.Context)
+            {
+                teamEntity = context.Teams.Find(_team.Id);
+                if (teamEntity != null)
+                {
+                    roster = teamEntity.Players.Select(x => new PlayerId(x.Id));
+                }
+            }
+
+            teamEntity.Should().NotBe(null, teamShouldBeSavedToDb);
+
+            var teamFromDb = new Team(teamEntity.Id, teamEntity.Name, teamEntity.Coach, teamEntity.Achievements, new PlayerId(teamEntity.CaptainId), roster);
+
+            teamComparer.AreEqual(teamFromDb, _team).Should().BeTrue();
+        }
+
+        [Then(@"Validation fails")]
+        public void ThenValidationFails()
+        {
+            if (_captain==null)
```

Comment:
What is intent of first part of if? How it relates to asserting that validation failed?